### PR TITLE
Fix errors thrown when picking terrain

### LINKED ISSUE #3779: throwOnDestroy errors when terrain and picking are enabled
If you go to a demo that has terrain enabled and is constantly picking, such as Ground Primitive, you eventually start to get `This object was destroyed` errors if you move the mouse around enough. I traced this back to https://github.com/AnalyticalGraphicsInc/cesium/commit/71bcd26ea9d81344bacb98bd03ac52e5ebc74d65 @bagnell.

@tfili This is not 3d-tiles related after all. The 3d tiles demo is also picking every frame which exposed this problem.

### PR DESCRIPTION
**Pull request by bagnell**
Fixes #3779.

The `GlobeSurfaceTileProvider` keeps a list of commands submitted for rendering. Let one of the commands contain a vertex array created from upsampling. After executing the draw calls but before rendering finishes, the tiles are processed (loading/creating/transforming imagery/terrain). During the processing, its possible that an upsampled terrain tile can receive its actual data. The upsampled vertex array will be destroyed and replaced with the new mesh. If a pick happens right after a render, the list of commands now has a destroyed vertex array from the upsampled mesh.

There was a couple of fixes that I tried, but they all failed because some places in the code expect the commands executed the previous frame to be able to execute again. This was just the first place it would fail. The solution was to reference count the vertex arrays.

__TLDR;__ We should reference count all WebGL resources.

### CODE REVIEW

**Comment by mramato** [comment]
> TLDR; We should reference count all WebGL resources.

:+1: Totally agree with this (and the `destroy` patter we currently use as a whole should be replaced with reference counting.)

**Comment by pjcozzi** [comment]
This fix seems awkward.  Maybe it is the only approach, but can you walk me through it today along with the others that you tried.

**Comment by bagnell** [comment]
@pjcozzi This is ready for another look. 

After our dscussion offline, I went back and changed it so that vertex arrays are queued to be destroyed at the start of the next frame. It only happens for tiles that are updated with new data. Vertex arrays of tiles that are being unloaded still happen immediately. It wasn't as bad as I thought.

**Comment by pjcozzi** [comment]
This is a bit cleaner and more direct.

Update CHANGES.md.

Is it possible to add a reasonable test?  Or is it not worth it?

**Comment by bagnell** [comment]
@pjcozzi This is ready. I don't think I could reproduce it in a unit test. The only way I could reproduce the problem was to pick on mouse move and constant camera movement.
